pandas/core/arrays/datetimes.py

```diff
@@ -794,16 +794,17 @@ def _add_offset(self, offset):
                 values = self.tz_localize(None)
             else:
                 values = self
-            result = offset.apply_index(values)
-            if self.tz is not None:
-                result = result.tz_localize(self.tz)
+            result = offset.apply_index(values).tz_localize(self.tz)
 
         except NotImplementedError:
             warnings.warn(
                 "Non-vectorized DateOffset being applied to Series or DatetimeIndex",
                 PerformanceWarning,
             )
             result = self.astype("O") + offset
+            if len(self) == 0:
```

Comment:
use ``not len``

Reply:
follow-up, ive got a branch specifically collecting these things